### ISSUE TITLE
Nullness Checker/Initialization Checker issue with UnderInitialization

### ISSUE THREAD

**Issue by wmdietl** [typetools/checker-framework#779]
Let us consider two files with the Nullness Checker. 
InitZero is in the repository as TODO
Init is in the repository as TODO

InitZero is the version without annotations. 
Java compiles it without a warning, but executing it will result in an NPE: 

```
Exception in thread "main" java.lang.NullPointerException 
at B.foo(InitZero.java:17) 
at A.<init>(InitZero.java:5) 
at B.<init>(InitZero.java:13) 
at InitZero.main(InitZero.java:23) 
```

Compiling it with the Checker Framework and `-processor nullness` gives: 

```
InitZero.java:5: error: [method.invocation.invalid] call to foo() not 
allowed on the given receiver. 
foo(); 
  ^ 
  found   : @UnderInitialization(A.class) @NonNull A 
  required: @Initialized @NonNull A 
1 error 
```

This is the expected behavior and it prevents the invalid invocation 
of a virtual method from the constructor. 

However, I then went ahead and annotated the foo method with 
`@UnderInitialization(A.class)`.
I expected the warning on the invocation of foo to go away and instead 
expected a warning on the implementation of B.foo. 

However, again running with `-processor nullness`, there was not a 
single warning and Init.java compiles successfully. When executing the 
resulting program, one gets the same NPE again. 

This is with the 2.0.1 release version of the Checker Framework. 

[This issue was first discussed here: https://groups.google.com/d/msg/checker-framework-dev/YhNZvCE2lmw/w3NxDIgHBgAJ ]
